src/app/src/main/java/com/example/bookmark/BorrowerBookDetailsActivity.java

```diff
@@ -0,0 +1,288 @@
+package com.example.bookmark;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.ImageView;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.example.bookmark.abstracts.ListingBooksActivity;
+import com.example.bookmark.models.Book;
+import com.example.bookmark.models.Request;
+import com.example.bookmark.models.User;
+import com.example.bookmark.server.StorageServiceProvider;
+import com.example.bookmark.util.DialogUtil;
+import com.example.bookmark.util.RequestUtil;
+
+/**
+ * This activity shows the details of a book. As well, depending on the status of the book,
+ * the activity will configure the action button to prompt the user to execute different actions
+ * while also implementing different functionality for each status of the book.
+ *
+ * @author Nayan Prakash
+ */
+public class BorrowerBookDetailsActivity extends BackButtonActivity {
+
+    private static final String TAG = "Borrower Book Details";
+
+    private static final int SCAN_ISBN_TO_RETURN = 101;
+    private static final int SCAN_ISBN_TO_BORROW = 102;
+
+    String isbn;
+    String title;
+    String author;
+    String description;
+    String ownedBy;
+    String status;
+    // TODO: Image image;
```

Comment:
Just as a heads up, you should be able to store, retrieve, and delete Photograph objects to Firebase now.

Reply:
Once this is approved, merged, and tested i'll go through the whole app turning them on everywhere/debuggin